crates/ra_hir_ty/src/diagnostics.rs

```diff
@@ -296,7 +307,7 @@ mod tests {
 
         let mut actual: FxHashMap<FileId, Vec<(TextRange, String)>> = FxHashMap::default();
         db.diagnostics(|d| {
-            // FXIME: macros...
+            // FIXME: macros...
             let file_id = d.source().file_id.original_file(&db);
             let range = d.syntax_node(&db).text_range();
```

Comment:
> d.syntax_node(&db)

This is the only usage of this public method: https://github.com/rust-analyzer/rust-analyzer/blob/d2241f5b0e105055348fe626e4da6b1a4e24be58/crates/ra_hir_expand/src/diagnostics.rs#L42-L52
in RA codebase.

I think with the two separate elements for a diagnostics, the method name currently might cause some confusion, should we remove it?

Reply:
yeah, it's better to remove it

Reply:
Removed the whole `impl dyn Diagnostic` eventually.